app/db.js

```diff
@@ -7,42 +7,42 @@ const nconf = require('../config');
 const dbUrl = nconf.get('db');
 mongoose.Promise = Promise;
 
+let tearingDown = false;
+
 const connect = function () {
-  /**
-   const options = {
-    server: {
-      socketOptions: {keepAlive: 1},
-      auto_reconnect: true
-    }
-  }; */
   const options = {
     useMongoClient: true,
-    // logger: logger,
-    loggerLevel: 'warning' // @todo fetch from config file
+    logger: logger.log.bind(logger),
+    loggerLevel: 'silly' // @todo fetch from config file
   };
-  logger.info(`Create MongoDB connection: ${dbUrl}`);
+
+  logger.info(`Connecting to MongoDB: ${dbUrl}`);
   return mongoose
-    .connect(dbUrl, options)
-    .then(() => {
-      mongoose.connection.on('error', () => {
-        logger.error(`Could not connect to MongoDB: ${dbUrl}`);
-      });
-    });
+    .connect(dbUrl, options);
 };
 
 const close = Promise.promisify(mongoose.connection.close.bind(mongoose.connection));
 function disconnect() {
+  tearingDown = true;
   logger.info(`Force to close the MongoDB connection: ${dbUrl}`);
   return close();
 }
 
+mongoose.connection.on('error', (error) => {
+  logger.error(`MongoDB connection error: ${error.message}`);
+});
+
 mongoose.connection.on('disconnected', () => {
-  logger.warn('MongoDB connection lost, try again');
-  connect();
```

Comment:
so mongoose by default reconnects?

Reply:
That's the image I got from the documentation, seemed to work when I manually crashed and brought up the mongo process a few times.